core/src/main/java/org/apache/calcite/rel/rel2sql/RelToSqlConverter.java

```diff
@@ -828,6 +830,27 @@ public Result visit(Uncollect e) {
     return result(asNode, ImmutableList.of(Clause.FROM), e, null);
   }
 
+  public Result visit(TableFunctionScan e) {
+    final List<SqlNode> inputSqlNodes = new ArrayList<>();
+    for (int i = 0; i < e.getInputs().size(); i++) {
+      Result child = visitChild(i, e.getInput(i));
+      inputSqlNodes.add(child.asStatement());
```

Comment:
You can get size of inputs of `TableFunctionScan` in advance, which can help initialize the array list. Besides, it is better to use `for (int i = 0; i < size; i++) ` instead of `for (int i = 0; i < e.getInputs().size(); i++)`.

Reply:
Sure, that would be better.
Thanks @chunweilei .